src/xacro/__init__.py

```diff
@@ -104,7 +116,7 @@ def load_yaml(filename):
     f = open(filename)
     oldstack = push_file(filename)
     try:
-        return yaml.safe_load(f)
+        return YamlDictWrapper(yaml.safe_load(f))
```

Comment:
Could YamlDictWrapper be exposed for general use in Xacro files? Why restrict the dot notation to just Yaml-loaded files? Why not allow any dictionary to be wrapped with this to enable read-only dotted access? For example:

```
<xacro:property name="dictionary" value="${dict(a=1, b=2}"/>
<xacro:property name="dotted" value="${YamlDictWrapper(dictionary)}"/>
```

However, if I try this directly, I get an error:
` name 'YamlDictWrapper' is not defined`

To make it even more user-friendly, perhaps a function 'dotify' could be exposed as below, that wraps the given dictionary within YamlDictWrapper?
`<xacro:property name="dotted" value="${dotify(dictionary)}"/>
`

Reply:
That should be possible. Other opinions, @gavanderhoorn, particularly about the function name `dotify`?

Reply:
Hm.

I'm not sure about the name `dotify`. Seems to be meaningful only if you already know what "dotted access" is.

Can't really come up with anything better right now though.

Also: to avoid breaking existing user's setups: would it make sense to add some sort of prefix? I'm not sure whether that's actually a risk, but seeing as `xacro` is such a common dependency, it would be good to avoid the possibility.

Reply:
I'm not good at inventing new names either. What kind of prefix do you think of? Something like `xacro.dotify`?
And we warn/throw if the user attempts to override the namespace `xacro`?

Reply:
> Something like `xacro.dotify`?

yes, that might be nice actually.

> And we warn/throw if the user attempts to override the namespace `xacro`?

that would seem prudent, yes.

Reply:
I'm hesitant to introduce a `xacro` package namespace. Actually, we already have a safety check issuing a warning if any of our global symbols is overridden: 
https://github.com/ros/xacro/blob/a4f0b84e166de5742f6ac83c561d1e5b92eb3ba7/src/xacro/__init__.py#L325-L326

As overriding works, it will not break existing xacros as well (just issue the warning).